packages/airnode-node/src/adapters/http/worker.ts

```diff
@@ -9,17 +9,21 @@ import {
   WorkerFunctionName,
   WorkerOptions,
 } from '../../types';
+import { CallApiArgs } from '../../workers/local-handlers';
+import type { ApiCallOptions } from '../../handlers/call-api';
 
 export async function spawnNewApiCall(
   aggregatedApiCall: AggregatedApiCall,
   logOptions: LogOptions,
   workerOpts: WorkerOptions,
-  encodeResponse = true
+  apiCallOptions?: ApiCallOptions
 ): Promise<LogsData<ApiCallResponse | null>> {
   const options = {
     ...workerOpts,
     functionName: 'callApi' as WorkerFunctionName,
-    payload: { aggregatedApiCall, logOptions, encodeResponse },
+    // TODO: The payload should have the same type for all handlers (local and AWS). Currently it does not.
+    // AWS handlers (in deployer) expects the `config` property to be present in the payload.
```

Comment:
WDYT? Did I miss something?

Reply:
I'm not sure I understand, sorry. The AWS handlers in the deployer don't look like they expect the `config` object to be present in the payload.

https://github.com/api3dao/airnode/blob/b152f7e3426aa13e3a12e9ec616036642ecdcba3/packages/airnode-deployer/src/handlers/aws/index.ts#L18
https://github.com/api3dao/airnode/blob/b152f7e3426aa13e3a12e9ec616036642ecdcba3/packages/airnode-deployer/src/handlers/aws/index.ts#L41
https://github.com/api3dao/airnode/blob/b152f7e3426aa13e3a12e9ec616036642ecdcba3/packages/airnode-deployer/src/handlers/aws/index.ts#L5

Reply:
Yeah, thanks. I was confused because actually the I've broken things with this refactor - I was not passing the `config` correctly and I was confused why is not passed anywhere...

Anyway, I'll separate the refactor part out of this PR and try to do in in a separate branch (and PR).